crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -578,12 +578,13 @@ impl Analyzer<'_, '_> {
                                 return Ok(ty);
                             }
 
-                            let ty = self
-                                .normalize(span, Cow::Owned(prop_ty), opts)
-                                .context("tried to normalize the type of property")?
-                                .into_owned();
-
-                            return Ok(Cow::Owned(ty));
+                            // TODO(): scope of normalize should be obj_ty scope
```

Comment:
Other than with_child, is there a better way to specify the scope as that object?

Reply:
You mean you only want to change `scope.this`?

Reply:
If so, I simply modifed `self.scope.this` like https://github.com/dudykr/stc/blob/81792e8e9dcfd95c4e3e0b98001a8ec47e0af27c/crates/stc_ts_file_analyzer/src/analyzer/class/mod.rs#L2461

Reply:
If you want to ensure that you don't forget resetting `self.scope.this`, you can do like

```rust

let old_this = self.scope.this.take();
self.scope.this = new_this;
let result = (|| {
    // your code.
})();
self.scope.this = old_this;

```

Reply:
I want a scope outside of the traditional context.
Like when you call a closure, the closure doesn't use the caller's scope.

Of course, it's fine to do it now or with the code you suggested for now.